presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -158,27 +162,59 @@ private void validateShowStatsSubquery(ShowStats node, Query query, QuerySpecifi
+        private Node rewriteShowStats(ShowStats node, Table table, List<SelectItem> selectItems, Constraint constraint)
         {
+            List<String> statsColumnNames = buildColumnsNames();
             TableHandle tableHandle = getTableHandle(node, table.getName());
             TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, constraint);
-            List<String> statsColumnNames = buildColumnsNames();
-            List<SelectItem> selectItems = buildSelectItems(statsColumnNames);
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+
             Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
-            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, tableStatistics);
+            Multimap<String, String> extractNamedResultColumns = extractNamedResultColumns(node, tableMetadata, selectItems);
+            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, extractNamedResultColumns, tableStatistics);
 
-            return simpleQuery(selectAll(selectItems),
+            return simpleQuery(selectAll(buildSelectItems(statsColumnNames)),
                     aliased(new Values(resultRows),
                             "table_stats_for_" + table.getName(),
                             statsColumnNames));
         }
 
+        private Multimap<String, String> extractNamedResultColumns(ShowStats node, TableMetadata metadata, List<SelectItem> selectItems)
+        {
+            ImmutableMultimap.Builder<String, String> assignments = ImmutableMultimap.builder();
+
+            for (SelectItem item : selectItems) {
+                if (item instanceof AllColumns) {
+                    for (ColumnMetadata column : metadata.getColumns()) {
+                        if (!column.isHidden()) {
+                            assignments.put(column.getName(), column.getName());
+                        }
+                    }
+                }
+
+                if (item instanceof SingleColumn) {
+                    SingleColumn column = (SingleColumn) item;
+                    Optional<Identifier> alias = column.getAlias();
+                    Expression expression = column.getExpression();
+
+                    // TODO: Better error?
+                    check(expression instanceof Identifier, node, "Only table columns names are supported in SELECT");
+
+                    Identifier id = (Identifier) expression;
+                    if (alias.isPresent()) {
+                        assignments.put(id.getValue(), alias.get().getValue());
+                    }
+                    else if (column.getExpression() instanceof Identifier) {
+                        assignments.put(id.getValue(), id.getValue());
+                    }
+                }
+            }
+
+            return assignments.build();
```

Comment:
Add a test cases where two columns are aliased to same alias:
```
nationkey AS a, regionkey AS a
```

and same column

```
nationkey AS a, nationkey AS a
```

both should be disallowed.

Reply:
Actually i'd drop support for aliases for now.
For pretending we support real simple select items, we should also support identifier columns:
```
SHOW STATS FOR (SELECT nation.nationkey FROM nation)
```

and table aliases

```
SHOW STATS FOR (SELECT t.nationkey FROM nation AS t)
```

-- let's not do this now & keep things as simple as only needed for the permission check

In the long term, we should change the `SHOW STATS FOR (subquery)` so that it allows _any_ subquery. This will be slight semantics change, so the less we support now, the better.